gui/wxpython/gui_core/goutput.py

```diff
@@ -473,54 +465,35 @@ def OnCmdProgress(self, event):
         self.progressbar.SetValue(event.value)
         event.Skip()
 
-    def CmdProtocolSave(self):
-        """Save list of manually entered commands into a text log file"""
-        if self.cmdFileProtocol is None:
-            return  # it should not happen
-
-        try:
-            with open(self.cmdFileProtocol, "a") as output:
-                cmds = self.cmdPrompt.GetCommands()
-                output.write("\n".join(cmds))
-                if len(cmds) > 0:
-                    output.write("\n")
-        except IOError as e:
-            GError(
-                _("Unable to write file '{filePath}'.\n\nDetails: {error}").format(
-                    filePath=self.cmdFileProtocol, error=e
+    def OnCmdExportHistory(self, event):
+        """Export the history of executed commands stored
+        in a .wxgui_history file to a selected file."""
+        dlg = wx.FileDialog(
+            self,
+            message=_("Save file as..."),
+            defaultFile="grass_cmd_log.txt",
+            wildcard=_("{txt} (*.txt)|*.txt|{files} (*)|*").format(
+                txt=_("Text files"), files=_("Files")
+            ),
+            style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT,
+        )
+
+        if dlg.ShowModal() == wx.ID_OK:
+            path = dlg.GetPath()
+            try:
+                self.cmdPrompt.CopyHistory(path)
+            except IOError as e:
```

Comment:
I would move the exception handling in the CopyHistory. Check if the error type is appropriate, it seems OSError should be used instead.

Reply:
Good, I moved it to the CopyHistory. I think that the error type is okay according to https://docs.python.org/3/library/shutil.html : _Changed in version 3.3: [IOError](https://docs.python.org/3/library/exceptions.html#IOError) used to be raised instead of [OSError](https://docs.python.org/3/library/exceptions.html#OSError). Added follow_symlinks argument. Now returns dst._
But when I looked at the examples of how to use shutil.copyfile I found out that they use handling of different error types: https://www.geeksforgeeks.org/python-shutil-copyfile-method/ 
So difficult to decide... I would probably keep the OSError or just put the general `except Exception as e:` . What do you think?